timemachine/fe/atom_mapping.py

```diff
@@ -51,6 +58,36 @@ def mcs_map(a, b, threshold: float = 2.0, timeout: int = 5, smarts: Optional[str
     return rdFMCS.FindMCS([a, b], params)
 
 
+def mcs_map(a, b, threshold: float = 2.0, timeout: int = 5, smarts: Optional[str] = None):
```

Comment:
Need to refactor + sort out some naming: 
* This reseeding fallback should be equally applicable to the conformer-aware default `mcs` and to the conformer-oblivious `mcs_map_graph_only_complete_rings`, but currently I only applied it to one... Should I factor out this logic so that it can be used with arbitrary `mcs_fxn(Mol, Mol) -> MCSResult`?
* Can we remove the word `_map` and use similar levels of descriptiveness for the two `mcs_fxns` in this file? (`mcs_map` -> `mcs_conformer_aware`, `mcs_map_graph_only_complete_rings` -> `mcs_graph_only_complete_rings`) "`_map`" seems misleading here since these don't return atom maps, but RDKit `MCSResult`s...

Reply:
Applied name change and factored out `possibly_fallback_to_heavy_atom_mcs`, but I didn't immediately find a clean way to implement using `functools.wraps` or similar.

Current refactoring looks like:
https://github.com/proteneer/timemachine/blob/5c7a0379ee086a442eff7191b232e6a2981c8066/timemachine/fe/atom_mapping.py#L104-L126

This is not ideal -- now introduces two copies of each MCS method -- requiring docstrings to be either duplicated or less informative, and requiring the signature to be partially duplicated...

Reply:
@mcwitt -- is there a better way to do this off the top of your head? Seems like this should have a neater solution using something in `functools`...

Reply:
Simplified by consolidating the two graph-only and conformer-aware mcs fxns in this file and inlining this check -- may need to refactor again if we reintroduce multiple mcs fxns in this file